sharding/interface.go

```diff
@@ -181,6 +182,13 @@ type GenesisNodesSetupHandler interface {
 	IsInterfaceNil() bool
 }
 
+// NodeTypeProviderHandler defines the actions needed for a component that can handle the node type
+type NodeTypeProviderHandler interface {
```

Comment:
this can easily be redefined in the packages where used. As done in this PR it will only add an unnecessary dependency on the sharding package.

Reply:
done